File: Project_Euler/3-Largest_Prime_Factor/3-Largest_Prime_Factor.js
"use strict";

let number = 600851475143;
let max_prime = 0;
let i = 3;

while (number != 1) {
  if (number % i) i += 2;
  else {
    max_prime = i;
    number = number / i;
  }
}

console.log(max_prime);

File: Project_Euler/3-Largest_Prime_Factor/3-Largest_Prime_Factor.py
from math import *

def is_prime(x):
    if x == 2:
        return True
    if x < 2 or x % 2 == 0:
        return False

    for y in range(3, ceil(sqrt(x)) + 1, 2):
        if x % y == 0:
            return False

    return True

def largest_prime_factor(number):
    for x in range(ceil(sqrt(number)), 0, -1):
        if number % x == 0 and is_prime(x):
            return x
    return None

print(largest_prime_factor(600851475143))
